### DISCUSSION TITLE
Why can't Adguard Home itself be the DNS server

### DISCUSSION THREAD

**Discussion by blixten85** [AdguardTeam/AdGuardHome#4858] [Q&A]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to request a feature or enhancement and not ask a question

#### What problem are you trying to solve?
Why can't Adguard Home itself be the DNS server? 
Why rely on unbound or any other?

**Comment by fernvenue** [comment]
As @ameshkov said in https://github.com/AdguardTeam/AdGuardHome/issues/3160:

> AdGuard is a DNS forwarder, not a recursor.

That's why you must config some upstream servers in AdGuardHome.

> Regarding the recursor setup, if you are really sure that's what you need, consider running a local unbound instance and configure AGH to use it as an upstream. IMO, this does not provide any advantages, though. Mind that the queries that are made by a recursor are not encrypted so will be visible to your ISP.

By the way, in addition to the disadvantages mentioned above, recursive queries usually have efficiency problems when used in small areas, each query will take a very long time.
